pkg/generate/app/sourcelookup.go

```diff
@@ -378,7 +381,7 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		Binary: repo.binary,
 	}
 
-	if repo.Info() != nil && repo.Info().Dockerfile != nil {
+	if (repo.ignoreRepository || repo.forceAddDockerfile) && repo.Info() != nil && repo.Info().Dockerfile != nil {
```

Comment:
> All the bool does for now, if you think about this PR + #6615 together, is to hint us into adding the Dockerfile contents, however and wherever it came from (because, sadly, it is quite hard to reason about it).

if it's a hint as to whether or not we should add the dockerfile content, then @liggitt's inlineDockerfile (or addDockerfileToBuildConfig) makes perfect sense.  "dockerfile" tells me nothing.